tests/utils/utils.go

```diff
@@ -538,3 +490,70 @@ func DeployBatchCrossChainMessenger(
 
 	return address, exampleMessenger
 }
+
+func runExecutable(
+	cmd *exec.Cmd,
+	ctx context.Context,
+	appName string,
+	healthCheckUrl string,
+) chan struct{} {
+	cmdOutput := make(chan string)
+
+	// Set up a pipe to capture the command's output
+	cmdStdOutReader, err := cmd.StdoutPipe()
+	Expect(err).Should(BeNil())
+	cmdStdErrReader, err := cmd.StderrPipe()
+	Expect(err).Should(BeNil())
+
+	// Start the command
+	log.Info("Starting executable", "appName", appName)
+	err = cmd.Start()
+	Expect(err).Should(BeNil())
+
+	readyChan := make(chan struct{})
+
+	// Start goroutines to read and output the command's stdout and stderr
+	go func() {
+		scanner := bufio.NewScanner(cmdStdOutReader)
+		for scanner.Scan() {
+			log.Info(scanner.Text())
+		}
+		cmdOutput <- "Command execution finished"
+	}()
+	go func() {
+		scanner := bufio.NewScanner(cmdStdErrReader)
+		for scanner.Scan() {
+			log.Error(scanner.Text())
+		}
+		cmdOutput <- "Command execution finished"
+	}()
+	go func() {
+		err := cmd.Wait()
+		// Context cancellation is the only expected way for the process to exit, otherwise log an error
+		// Don't panic to allow for easier cleanup
+		if !errors.Is(ctx.Err(), context.Canceled) {
+			log.Error("Executable exited abnormally", "appName", appName, "err", err)
+		}
+	}()
+	go func() { // wait for health check to report healthy
+		for {
+			resp, err := http.Get(healthCheckUrl)
+			if err == nil && resp.StatusCode == 200 {
+				close(readyChan)
+				break
+			}
+			time.Sleep(time.Second * 1)
+		}
+	}()
+	return readyChan
+}
+
+// Helper function that waits for a signaling channel to be closed
+// or throws an error if the channel is not closed in time
+func WaitForChannelClose(ch <-chan struct{}, timeout time.Duration) {
```

Comment:
Can we pass a context instead, and leave it up to the caller to set a timeout on it? That's more canonical Go, and makes this function more flexible.

Reply:
We can, the reason I implemented it this way was to allow for simultaneous starts of multiple services which is what we do in one place so far but I can see it being done in more places in the future.

Reply:
Sorry I misunderstood what you meant. Yeah I will make the change